client/src/index.js

```diff
@@ -22,15 +23,25 @@ function Horizon({
   lazyWrites = false,
   authType = 'unauthenticated',
   keepalive = 60,
+  gaeInstanceIp = undefined, // undefined, 'external', 'internal'
   WebSocketCtor = WebSocket,
 } = {}) {
   // If we're in a redirection from OAuth, store the auth token for
   // this user in localStorage.
 
+  let wsHost = host
```

Comment:
I think this logic should move out of the client, since it's specific to a particular vendor. It's easy enough to add this to a wrapper library